pkg/core/resources/apis/mesh/mesh_helpers.go

```diff
@@ -70,3 +74,40 @@ func (m *MeshResource) GetCertificateAuthorityBackend(name string) *mesh_proto.C
 	}
 	return nil
 }
+
+var durationRE = regexp.MustCompile("^([0-9]+)(y|w|d|h|m|s|ms)$")
+
+// ParseDuration parses a string into a time.Duration
+func ParseDuration(durationStr string) (time.Duration, error) {
+	// Allow 0 without a unit.
+	if durationStr == "0" {
+		return 0, nil
+	}
+	matches := durationRE.FindStringSubmatch(durationStr)
+	if len(matches) != 3 {
```

Comment:
Will this allow me to do `1y2w3d4h5m6s`?

Reply:
no @nickolaev

Reply:
Will it make sense to iterate over the matches slice, so such formats can be supported?

Reply:
yes, we can, but thinking that won't be the optimal solution for this.

Reply:
Yes, I agree. Let's merge this one.